packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,44 @@
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+import { openseaAssetsApiUrl } from '../constants'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const tokenDetailsResponse = useAsync(
+        async function () {
+            let updatedTokenDetails: TokenDetails = {}
+            const [address, tokenId] = parseNftUrl(props.nftUrl)
+
+            // if nftUrl is invalid
+            if (!(address && tokenId)) return updatedTokenDetails
+
+            const link = new URL(openseaAssetsApiUrl)
+            link.searchParams.set('asset_contract_address', address)
+            link.searchParams.set('token_ids', tokenId)
+
+            try {
+                const response = await Services.Helper.fetch(link.toString())
+                const { assets } = JSON.parse(await response.text())
```

Comment:
https://mdn.io/Response.prototype.json

Reply:
@mindey plz use `response.json()` replace your here line code.

Reply:
`Services.Helper.fetch` returns `x.blob()` instead of `x.json()`, so we will need to do a `JSON.parse` to convert this blob into json. I think we can resolve this and later update the `Services.Helper.fetch` function and all usages.

@septs what do you think? I have also resolved the other two reviews.

```
export function fetch(url: string) {
    return globalThis.fetch(url).then((x) => x.blob())
}
```

Reply:
CC @Jack-Works

Reply:
Let's leave it as-is. `Response` is not serializable so I choose to make it a `Blob`. We may refactor it to other forms in the future.